src/components/Swap/SwapActionButton/index.tsx

```diff
@@ -16,17 +19,29 @@ export default function SwapActionButton() {
     [Field.INPUT]: { currency: inputCurrency, amount: inputCurrencyAmount, balance: inputCurrencyBalance },
     [Field.OUTPUT]: { currency: outputCurrency },
     error,
+    approval,
+    allowance,
     trade: { trade },
   } = useSwapInfo()
+  const permit2Enabled = usePermit2()
   const isWrap = useIsWrap()
-  const isDisabled = useMemo(
-    () =>
+  const isDisabled = useMemo(() => {
+    if (permit2Enabled) {
```

Comment:
These are booleans just like the rest of the list - include them in the compound conditional statement, instead of making new expressions.

Reply:
This belongs in #357, otherwise #357 breaks this behavior.